spec/que/poller_spec.rb

```diff
@@ -275,14 +277,15 @@ def assert_poll(priorities:, locked:)
       assert_equal false, poller.should_poll?
     end
 
-    it "should be false if the number of jobs returned from the last poll was less than the lowest priority request, but the poll_interval has elapsed" do
+    it "should be true if the number of jobs returned from the last poll was less than the lowest priority request, but the poll_interval has elapsed" do
       job_ids = 5.times.map { Que::Job.enqueue.que_attrs[:id] }
 
       result = poller.poll(priorities: { 500 => 7 }, held_locks: Set.new)
       assert_equal job_ids, result.map(&:id)
 
-      poller.instance_variable_set(:@last_polled_at, Time.now - 30)
-      assert_equal true, poller.should_poll?
+      Timecop.freeze(Time.now + 30) do
+        assert_equal true, poller.should_poll?
+      end
```

Comment:
This change makes the spec less coupled with implementation details.

Instead of setting private instance variables, it 'time travels' 30 seconds into the future, after the `poll_interval` has already elapsed.